internal/provider/organization_template_data_resource.go

```diff
@@ -0,0 +1,143 @@
+			"Unexpected Organization Template Data Source Configure Type",
+			fmt.Sprintf("Expected *TerrakubeConnectionData got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	if providerData.InsecureHttpClient {
+		if custom, ok := http.DefaultTransport.(*http.Transport); ok {
+			customTransport := custom.Clone()
+			customTransport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+			d.client = &http.Client{Transport: customTransport}
+		} else {
+			d.client = &http.Client{}
+		}
+	} else {
+		d.client = &http.Client{}
+	}
+	d.endpoint = providerData.Endpoint
+	d.token = providerData.Token
+
+	ctx = tflog.SetField(ctx, "endpoint", d.endpoint)
+	ctx = tflog.SetField(ctx, "token", d.token)
+	ctx = tflog.MaskFieldValuesWithFieldKeys(ctx, "token")
+	tflog.Info(ctx, "Organization Template Data Source configured")
+}
+
+func (d *OrganizationTemplateDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_organization_template"
+}
+
+func (d *OrganizationTemplateDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "Id",
+			},
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "Organization Template Name",
+			},
+			"organization_id": schema.StringAttribute{
+				Required:    true,
+				Description: "Organization ID",
+			},
+		},
+	}
+}
+
+func (d *OrganizationTemplateDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state OrganizationTemplateDataSourceModel
+
+	req.Config.Get(ctx, &state)
+
+	apiUrl := fmt.Sprintf("%s/api/v1/organization/%s/template?filter[template]=name=='%s'", d.endpoint, state.OrganizationId.ValueString(), url.PathEscape(state.Name.ValueString()))
+	reqTemplate, err := http.NewRequest(http.MethodGet, apiUrl, nil)
+	reqTemplate.Header.Add("Authorization", fmt.Sprintf("Bearer %s", d.token))
+	reqTemplate.Header.Add("Content-Type", "application/vnd.api+json")
+	if err != nil {
+		tflog.Error(ctx, "Error creating organization template datasource request")
+	}
+
+	resTemplate, err := d.client.Do(reqTemplate)
+	if err != nil {
+		tflog.Error(ctx, "Error executing organization template datasource request")
+	}
+
+	body, err := io.ReadAll(resTemplate.Body)
+	if err != nil {
+		tflog.Error(ctx, "Error reading organization response")
+	}
+
+	var templates []interface{}
+
+	templates, err = jsonapi.UnmarshalManyPayload(strings.NewReader(string(body)), reflect.TypeOf(new(client.OrganizationTemplateEntity)))
+
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to unmarshal payload", fmt.Sprintf("Unable to marshal, error: %s, response status %s, reponse body %s", err, resTemplate.Status, string(body)))
```

Comment:
The build is failing with this error.

```
run golangci-lint
  Running [/home/runner/golangci-lint-1.59.1-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
  Error: `reponse` is a misspelling of `response` (misspell)
```

I think it for that `"reponse"`

Reply:
Thanks, updated